### PR TITLE
Improved names used in charmbuilder script to reflect better the reality.

### PR DESCRIPTION
**Pull request by facundobatista**

### CODE REVIEW

File: charmcraft/charm_builder.py [hunk 1]
```diff
@@ -365,13 +365,13 @@ def _parse_arguments() -> argparse.Namespace:
         help="The charm entry point. Default is 'src/charm.py'.",
     )
     parser.add_argument(
-        "--charmdir",
+        "--builddir",
```
Comment (mr-cal) [inline charmcraft/charm_builder.py:368]: How much will this affect the end user?

I wonder if it is worth having a deprecation notice or notifying users to use the new arguments.  The `--builddir` parameter keeping the same name but changing its meaning could be confusing.

Comment (mr-cal) [reply charmcraft/charm_builder.py:368]: Discussed offline - `charm_builder` isn't called by the user.

Comment (facundobatista) [reply charmcraft/charm_builder.py:368]: no final user involved! :) this is deep in the execution layers of `pack` command

**Review by mr-cal — APPROVED** [review]
+1

**Pull request merged**